packages/kbn-palettes/classes/color_fn_palette.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Optional } from 'utility-types';
+import { KbnBasePaletteConfig, KbnBasePalette } from './palette';
+import { IKbnPalette, KbnPaletteType } from './types';
+
+const DEFAULT_COLOR_COUNT = 10;
+
+export interface KbnColorFnPaletteConfig extends Optional<KbnBasePaletteConfig, 'colorCount'> {
+  type: KbnPaletteType;
+  colorFn: (n: number) => string[];
+  defaultNumberOfColors?: number;
+}
+
+export class KbnColorFnPalette extends KbnBasePalette implements IKbnPalette {
+  public readonly type: KbnPaletteType;
+
+  #colorFn: (n: number) => string[];
+  #defaultNumberOfColors: number;
+
+  constructor({
+    type,
+    colorFn,
+    defaultNumberOfColors,
+    colorCount = DEFAULT_COLOR_COUNT,
+    ...rest
+  }: KbnColorFnPaletteConfig) {
+    super({ ...rest, colorCount });
+
+    this.type = type;
+
+    this.#colorFn = colorFn;
+    this.#defaultNumberOfColors = defaultNumberOfColors ?? colorCount;
+  }
+
+  public colors = (n: number = this.#defaultNumberOfColors) => {
+    return this.#colorFn(n);
```

Comment:
Should we prevent bad numbers here? negative, non-finite?

Reply:
Sure, I enforcement for min of `1`. See 016f1615074a66354d91174004f5b9ea31ef4a56